### ISSUE TITLE
[Question] What is the best way to parallelize on the graph level

### ISSUE THREAD

**Issue by CJ-Wright** [mrocklin/streamz#58]
The dask extensions have given us the ability to parallelize on the "inside a node" level. However, some nodes can be run completely independently from one another. What is the best way to access that level of parallelizem?

**Comment by mrocklin** [comment]
I'll go ahead and claim that the current approach exposes all available parallelism.  We just emit tasks and the scheduler manages them as appropriate.

**Comment by CJ-Wright** [comment]
Cool. I guess I need to start playing around more with the dask stuff.

**Comment by mrocklin** [comment]
I'm confident that it will break on you in many obvious and subtle ways.
But in theory I think that it solves your problem just fine :)

**Comment by ordirules** [comment]
yeah, we've played with @mrocklin 's dask extension, passing futures. Seems to work well but we haven't rigorously tested yet. (but things like saving `Future`s to list and testing repeated computations are cached works very well. including `normalize_token` for arbitrary objects. so far so good for scientific computations!)
